src/game_api/entity.cpp

```diff
@@ -321,27 +321,28 @@ void Movable::set_position(float to_x, float to_y)
         rendering_info->x_dupe4 += dx;
         rendering_info->y_dupe4 += dy;
     }
-    if (State::get().ptr()->camera->focused_entity_uid == uid)
-        State::get().set_camera_position(dx, dy);
+    auto camera = HeapBase::get().state()->camera;
+    if (camera->focused_entity_uid == uid)
+        camera->set_position(dx, dy);
 }
 
 template <typename F>
-bool recursive(Entity* ent, std::optional<uint32_t> mask, std::vector<ENT_TYPE> ent_types, RECURSIVE_MODE rec_mode, F func)
+bool recursive(Entity* ent, std::optional<ENTITY_MASK> mask, std::vector<ENT_TYPE> ent_types, RECURSIVE_MODE rec_mode, F func)
 {
-    auto acutal_mask = [](uint32_t m) -> uint32_t // for the MASK.ANY
-    { return m == 0 ? 0xFFFF : m; };
+    auto actual_mask = [](ENTITY_MASK m) -> ENTITY_MASK // for the MASK.ANY
+    { return m == ENTITY_MASK::ANY ? (ENTITY_MASK)0xFFFF : m; };
 
     if (rec_mode == RECURSIVE_MODE::EXCLUSIVE)
     {
-        if (mask.has_value() && (acutal_mask(mask.value()) & ent->type->search_flags) != 0)
+        if (mask.has_value() && !!(actual_mask(mask.value()) & ent->type->search_flags))
```

Comment:
This approach `!!` is very debatable online, so I'm just highlighting it beforehand.
The only alternative I think is using function for it

Reply:
I don't think we have any rules on this, as long as it's readable and works well, I think it's fine